### PR TITLE
DOCSUP-10286: Update Join Table Engine documentation

### PR DESCRIPTION
**Pull request by olgarev**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation for #24822 and #23260

### CODE REVIEW

**Comment by olgarev** [comment]
@CurtizJ Hi, could you please review this PR?

I rearranged the sections because we always provide examples in the end, not in the beginning. And I changed one of the examples - removed the SETTINGS clause, because it generated an exception.

File: docs/en/engines/table-engines/special/join.md [hunk 1]
```diff
@@ -30,48 +30,82 @@ See the detailed description of the [CREATE TABLE](../../../sql-reference/statem
...
+
+`Join` table data is always located in the RAM. When inserting rows into a table, ClickHouse writes data blocks to the directory on the disk so that they can be restored when the server restarts.
+
+If the server restarts incorrectly, the data block on the disk might get lost or damaged. In this case, you may need to manually delete the file with damaged data.
+
+### Selecting and Inserting Data {#selecting-and-inserting-data}
+
+You can use `INSERT` queries to add data to the `Join`-engine tables. If the table was created with the `ANY` strictness, data for duplicate keys are ignored. With the `ALL` strictness, all rows are added.
+
+You cannot perform a `SELECT` query directly from the table. Instead, use one of the following methods:
```
Comment (vdimir) [inline docs/en/engines/table-engines/special/join.md:45]: It's not main use-case, but actually you can select data:

```
CREATE TABLE j (`a` String, `b` LowCardinality(String), `c` String) ENGINE = Join(ANY, LEFT, a);

insert into j values ('aa', 'vv', 'xx'), ('aa', '11', '22'), ('bb', 'AA', 'BB');

select * from j;

┌─a──┬─b──┬─c──┐
│ aa │ vv │ xx │
│ bb │ AA │ BB │
└────┴────┴────┘
```


```suggestion
Main use-cases for `Join`-engine tables are following:
```

File: docs/ru/engines/table-engines/special/join.md [hunk 2]
```diff
@@ -27,48 +27,82 @@ CREATE TABLE [IF NOT EXISTS] [db.]table_name [ON CLUSTER cluster]
...
+
+Данные таблиц `Join` всегда находятся в оперативной памяти. При вставке строк в таблицу ClickHouse записывает блоки данных в каталог на диске, чтобы их можно было восстановить при перезапуске сервера.
+
+При аварийном перезапуске сервера блок данных на диске может быть потерян или повреждён. В последнем случае может потребоваться вручную удалить файл с повреждёнными данными.
+
+### Выборка и добавление данных {#selecting-and-inserting-data}
+
+Для добавления данных в таблицы с движком `Join` используйте запрос `INSERT`. Если таблица создавалась со строгостью `ANY`, то данные с повторяющимися ключами игнорируются. Если задавалась строгость `ALL`, то добавляются все строки.
+
+Из таблиц нельзя выбрать данные с помощью запроса `SELECT`. Вместо этого, используйте один из следующих методов:
```
Comment (vdimir) [inline docs/ru/engines/table-engines/special/join.md:42]:
```suggestion
Основные применения `Join` таблиц:
```

File: docs/ru/engines/table-engines/special/join.md [hunk 3]
```diff
@@ -27,48 +27,82 @@ CREATE TABLE [IF NOT EXISTS] [db.]table_name [ON CLUSTER cluster]
...
+
+-   Используйте таблицу как правую в секции `JOIN`.
+-   Используйте функцию [joinGet](../../../engines/table-engines/special/join.md#joinget), которая позволяет извлекать данные из таблицы таким же образом как из словаря.
+
```
Comment (vdimir) [inline docs/ru/engines/table-engines/special/join.md:46]:
```suggestion
-   Использование в правой части секции `JOIN`.
-   Извлечение данных из таблицы таким же образом как из словаря с помощью функции [joinGet](../../../engines/table-engines/special/join.md#joinget).

```

File: docs/ru/engines/table-engines/special/join.md [hunk 4]
```diff
@@ -27,48 +27,82 @@ CREATE TABLE [IF NOT EXISTS] [db.]table_name [ON CLUSTER cluster]
...
+-   [max_rows_in_join](../../../operations/settings/query-complexity.md#settings-max_rows_in_join)
+-   [max_bytes_in_join](../../../operations/settings/query-complexity.md#settings-max_bytes_in_join)
+-   [join_overflow_mode](../../../operations/settings/query-complexity.md#settings-join_overflow_mode)
+-   [join_any_take_last_row](../../../operations/settings/settings.md#settings-join_any_take_last_row)
+-   [persistent](../../../operations/settings/settings.md#persistent)
+
+Таблицы с движком `Join` нельзя использовать в операциях `GLOBAL JOIN`.
+
+Движок `Join` позволяет использовать параметр [join_use_nulls](../../../operations/settings/settings.md#join_use_nulls) в запросе `CREATE TABLE`, который также можно использовать в запросе [SELECT](../../../engines/table-engines/special/join.md). Если у вас разные настройки `join_use_nulls`, вы можете получить сообщение об ошибке при объединении таблиц. Это зависит от типа соединения. Когда вы используете функцию [joinGet](../../../engines/table-engines/special/join.md#joinget), вам необходимо использовать один и тот же параметр `join_use_nulls` в запросах `CRATE TABLE` и `SELECT`.
```
Comment (vdimir) [inline docs/ru/engines/table-engines/special/join.md:64]:
```suggestion
Движок `Join` позволяет использовать настройку [join_use_nulls](../../../operations/settings/settings.md#join_use_nulls) в запросе `CREATE TABLE`. Необходимо использовать одно и то же значение параметра `join_use_nulls` в запросах `CRATE TABLE` и `SELECT`.
```

File: docs/en/engines/table-engines/special/join.md [hunk 5]
```diff
@@ -30,48 +30,82 @@ See the detailed description of the [CREATE TABLE](../../../sql-reference/statem
...
+-   [join_use_nulls](../../../operations/settings/settings.md#join_use_nulls)
+-   [max_rows_in_join](../../../operations/settings/query-complexity.md#settings-max_rows_in_join)
+-   [max_bytes_in_join](../../../operations/settings/query-complexity.md#settings-max_bytes_in_join)
+-   [join_overflow_mode](../../../operations/settings/query-complexity.md#settings-join_overflow_mode)
+-   [join_any_take_last_row](../../../operations/settings/settings.md#settings-join_any_take_last_row)
+-   [persistent](../../../operations/settings/settings.md#persistent)
+
+The `Join`-engine tables can’t be used in `GLOBAL JOIN` operations.
+
+The `Join`-engine allows use [join_use_nulls](../../../operations/settings/settings.md#join_use_nulls) setting in the `CREATE TABLE` statement. And [SELECT](../../../sql-reference/statements/select/index.md) query allows use `join_use_nulls` too. If you have different `join_use_nulls` settings, you can get an error joining table. It depends on kind of JOIN. When you use [joinGet](../../../sql-reference/functions/other-functions.md#joinget) function, you have to use the same `join_use_nulls` setting in `CRATE TABLE` and `SELECT` statements.
```
Comment (vdimir) [inline docs/en/engines/table-engines/special/join.md:67]:
```suggestion
The `Join`-engine allows to specify [join_use_nulls](../../../operations/settings/settings.md#join_use_nulls) setting in the `CREATE TABLE` statement. [SELECT](../../../sql-reference/statements/select/index.md) query should have the same `join_use_nulls` value.
```

**Review by vdimir — APPROVED** [review]

**Pull request merged**